### DISCUSSION TITLE
Google Assistant Integration

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#3406] [Ideas]
Hi there!

I recently purchased the Google Pixel Buds A-Series, which have a Google Assistant integration. I use them to listen to music while doing sports. In xDrip itself there is a feature that can read the BG values for you, which is very useful, but quite annoying while doing sports. The Google Assistant provides some shortcuts for applications like Adidas Running or Deezer to do things like starting a new activity or playing your favourite music.

Is there a way to integrate xDrip Plus to the Google Assistant to use some shortcuts for it, like "How is my current blood sugar" or "How much insulin is active"? I think you cam define the voice commands by yourself, but the app need to provide an Interface to the Google Assistant. 

I would be very pleased if you would implement this interface in coming versions of the app, I think it would make the life of sporty people easier, because they can ask for their BG whenever they want. 
That could be a useful extension to this really feature-rich app!

Best regards,
Miquest

**Comment by Navid200** [comment]
Can you give more detail what is wrong with the existing Speak Readings feature?
Before we can consider adding a new feature, it's best to go over the existing options and explain what they lack.

**Reply by Navid200** [reply]
We have an outstanding enhancement request to add audio focus for speak readings.  After we add that, if you are listening to music, you can have the volume decrease while your blood glucose is read or completely muted or no change.

**Comment by Miquest** [comment]
Hi!

Sorry that I wasn't able to answer earlier, I was really busy last days.  

First, I want to say that I don't think there's anything wrong with the existing Speak reading feature for the usage scenario it's meant for. 

What I meant is an improved version of the feature, where you can ask for your blood glucose whenever you want. 
While doing sports, you don't want to be disturbed every five minutes by the voice reading your current bg, I think that's quite annoying. Another scenario for the Google Assistant integration could be a smart home with Google Nest Hubs. I think many people would find it helpful if there would be the opportunity to ask their Nest Hub for their bg level when the phone is in the next room and they don't want to stop doing what they're doing in that moment.

Sure, it is not really necessary at all, but it could improve the user experience in cases I described before. Just an idea. 

You guys are doing absolutely fantastic work with this already fantastic application!
